File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/7 - 9 and 10 March 2019/04. Game Number Wars.py
name_player_one = input()
name_player_two = input()

player_one_points = 0
player_two_points = 0

command = None

while command != "End of game":
    command = input()
    if command == "End of game":
        break
    player_one_card = int(command)
    player_two_card = int(input())
    if player_one_card > player_two_card:
        player_one_points += abs(player_one_card - player_two_card)
    elif player_two_card > player_one_card:
        player_two_points += abs(player_two_card - player_one_card)
    elif player_one_card == player_two_card:
        final_one_card = int(input())
        final_two_card = int(input())
        if final_one_card > final_two_card:
            player_two_points = 0
        elif final_two_card > final_one_card:
            player_one_points = 0
        print("Number wars!")
        break

if command == "End of game":
    print(f"{name_player_one} has {player_one_points} points")
    print(f"{name_player_two} has {player_two_points} points")
else:
    if player_one_points > player_two_points:
        print(f"{name_player_one} is winner with {player_one_points} points")
    else:
        print(f"{name_player_two} is winner with {player_two_points} points")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/03. Online Exam - 9 and 10 March 2019/04. Game Number Wars.js
function gameNumberWars(input) {
    // Mask
    let playerOne = input.shift();
    let playerTwo = input.shift();

    let playerOnePoints = 0;
    let playerTwoPoints = 0;

    while ('I have farts!') {
        let command = input.shift();
        if (command == 'End of game') {
            console.log(`${playerOne} has ${playerOnePoints} points`)
            console.log(`${playerTwo} has ${playerTwoPoints} points`)
            break;  // And stop the Program.
        }

        let playerOneCard = Number(command);
        let playerTwoCard = Number(input.shift());

        if (playerOneCard > playerTwoCard) {playerOnePoints += playerOneCard - playerTwoCard;}
        else if (playerTwoCard > playerOneCard) {playerTwoPoints += playerTwoCard - playerOneCard;}
        else {
            // We have Number Wars.... 
            console.log("Number wars!");
            let finalPlayerOneCard = Number(input.shift());
            let finalPlayerTwoCard = Number(input.shift());

            if (finalPlayerOneCard > finalPlayerTwoCard) {
                console.log(`${playerOne} is winner with ${playerOnePoints} points`);
            } else {
                console.log(`${playerTwo} is winner with ${playerTwoPoints} points`);
            }
            break;  // And stop the Program.
        }
    }
}

gameNumberWars(['Desi', 'Niki', 7, 5, 3, 4, 3, 3, 5, 3]);  // Should return:
// Number wars!
// Desi is winner with 2 points

gameNumberWars(['Elena', 'Simeon', 6, 3, 2, 5, 8, 9, 'End of game']);  // Should return:
// Elena has 3 points
// Simeon has 4 points
